Typescript Error in MarkdownDialog

```
src/components/Markdown/MarkdownDialog.vue:268:14 - error TS2322: Type '(response: string) => void' is not assignable to type '(results: SelectionItem[]) => void'.
  Types of parameters 'response' and 'results' are incompatible.
    Type 'SelectionItem[]' is not assignable to type 'string'.
```

--- a/client/src/components/DataDialog/DataDialog.vue
+++ b/client/src/components/DataDialog/DataDialog.vue
@@ -44,7 +44,7 @@ const props = withDefaults(defineProps<Props>(), {
 
 const emit = defineEmits<{
     (e: "onCancel"): void;
-    (e: "onOk", results: Array<Record>): void;
+    (e: "onOk", results: unknown): void;
     (e: "onUpload"): void;
 }>();
 

--- a/client/src/components/Markdown/MarkdownDialog.vue
+++ b/client/src/components/Markdown/MarkdownDialog.vue
@@ -121,7 +121,7 @@ async function getHistories() {
     return data;
 }
 
-function onData(response: string) {
+function onData(response: unknown) {
     dataShow.value = false;
     emit("onInsert", `${props.argumentName}(history_dataset_id=${response})`);
 }

--- a/client/src/components/Markdown/MarkdownDialog.vue
+++ b/client/src/components/Markdown/MarkdownDialog.vue
@@ -252,7 +252,7 @@ if (props.argumentType == "workflow_id") {
             @onOk="onOk"
             @onCancel="onCancel" />
         <MarkdownVisualization
-            v-else-if="visualizationShow"
+            v-else-if="visualizationShow && currentHistoryId !== null"
             :argument-name="argumentName"
             :argument-payload="argumentPayload"
             :labels="effectiveLabels"
@@ -260,9 +260,14 @@ if (props.argumentType == "workflow_id") {
             :history="currentHistoryId"
             @onOk="onVisualization"
             @onCancel="onCancel" />
-        <DataDialog v-else-if="dataShow" :history="currentHistoryId" format="id" @onOk="onData" @onCancel="onCancel" />
+        <DataDialog
+            v-else-if="dataShow && currentHistoryId !== null"
+            :history="currentHistoryId"
+            format="id"
+            @onOk="onData"
+            @onCancel="onCancel" />
         <DatasetCollectionDialog
-            v-else-if="dataCollectionShow"
+            v-else-if="dataCollectionShow && currentHistoryId !== null"
             :history="currentHistoryId"
             format="id"
             @onOk="onDataCollection"

--- a/client/src/components/Markdown/MarkdownDialog.vue
+++ b/client/src/components/Markdown/MarkdownDialog.vue
@@ -29,6 +29,7 @@ const props = withDefaults(defineProps<MarkdownDialogProps>(), {
     argumentName: undefined,
     argumentType: undefined,
     argumentPayload: undefined,
+    labels: undefined,
 });
 
 const emit = defineEmits<{

